src/argilla/client/feedback/dataset.py

```diff
@@ -521,32 +543,84 @@ def push_to_argilla(
-                    " to Argilla."
-                )
-                return
-
             try:
-                for i in range(0, len(self.__new_records), PUSHING_BATCH_SIZE):
-                    datasets_api_v1.add_records(
-                        client=httpx_client,
-                        id=self.argilla_id,
-                        records=[
-                            record.dict()
-                            for record in tqdm(
-                                self.__new_records[i : i + PUSHING_BATCH_SIZE],
-                                desc="Pushing records to Argilla...",
-                                disable=show_progress,
-                            )
-                        ],
+                updated_records: List[FeedbackRecord] = []
+                for record in self.__records[:]:
+                    if record._updated and record.id is None:
+                        warnings.warn(
+                            "You are trying to update records that have not been fetched"
+                            " from Argilla yet and, so on, don't have an Argilla ID, which"
+                            " means that you may have pushed those in the same session"
+                            " and are trying to update them right after the `push_to_argilla`"
+                            " call. So on, you need to fetch those first with `fetch_records`"
+                            " before updating them, or just call `from_argilla`. For the"
+                            " moment the `suggestions` update will be ignored."
+                        )
+                        continue
+                    if record._updated:
+                        self.__records.remove(record)
+                        record._reset_updated()
+                        updated_records.append(record)
+
+                if len(self.__new_records) < 1 and len(updated_records) < 1:
+                    _LOGGER.warning(
+                        "Neither new records have been added nor existing records updated"
+                        " in the current `FeedbackDataset`, which means there are no"
+                        " records to push to Argilla."
                     )
+                    return
+
+                question_name2id = {question.name: question.id for question in self.__questions}
+                if len(updated_records) > 0:
+                    for i in trange(
+                        0,
+                        len(updated_records),
+                        PUSHING_BATCH_SIZE,
+                        desc="Updating records in Argilla...",
+                        disable=show_progress,
+                    ):
+                        for record in updated_records[i : i + PUSHING_BATCH_SIZE]:
+                            for suggestion in record.suggestions:
+                                suggestion.question_id = question_name2id[suggestion.question_name]
+                                datasets_api_v1.add_suggestion(
```

Comment:
since this is adding/updating a suggestion, maybe should be more descriptive calling `set_suggestion` instead of `add_suggestion`

Reply:
Since that's the SDK function, should we name it `put_suggestion` instead?

Reply:
Also, should we rename the `FeedbackRecord` method from `add_suggestions` to something else?

Reply:
For me, it makes sense to call `record.set_suggestion` since will set one single suggestion for a question. For the API call function, I would say not showing implementation details in the name (put_suggestions, or post_suggestions) we can adopt more general naming (set_suggestion, or update_suggestion, or replace_suggestion)